components/compass/button/_ button.scss

```diff
@@ -0,0 +1,220 @@
+            opacity: 0.12;
+        }
+        &::after {
+            opacity: 1;
+        }
+    }
+
+    &[disabled],
+    .disabled {
+        cursor: not-allowed;
+
+        &,
+        &:hover,
+        &.hover,
+        &:focus,
+        &.focus,
+        &:active,
+        &.active {
+            background: v(--center-channel-color-rgb, 0.1);
+
+            .Button_icon,
+            .Button_label {
+                color: v(--center-channel-color-rgb, 0.5);
+            }
+
+            &::before,
+            &::after {
+                opacity: 0;
+            }
+        }
+    }
+
+    // styles (primary, secondary, tertiary)
+    &.Button___primary:not([disabled]) {
+        color: v(--button-color-rgb);
+        background: v(--button-bg-rgb);
+
+        &:hover,
+        &.hover,
+        &:focus,
+        &.focus {
+            &::before {
+                opacity: 0.2;
+            }
+        }
+        &:focus,
+        &.focus {
+            &::after {
+                border-color: v(--link-color-rgb);
+                border-width: 2px;
+            }
+        }
+        &:active,
+        &.active {
+            &::before {
+                opacity: 0.32;
+            }
+        }
+    
+        .Button_icon {
+            color: v(--button-color-rgb);
+        }
+    }
+
+    &.Button___secondary:not([disabled]),
+    &.Button___tertiary:not([disabled]) {
+        color: v(--button-bg-rgb);
+        background: v(--button-color-rgb);
+
+        &::before {
+            background: v(--button-bg-rgb);
+        }
+        &::after {
+            border-color:v(--button-bg-rgb);
+        }
+
+        &:hover,
+        &.hover,
+        &:focus,
+        &.focus {
+            &:before {
+                opacity: 0.06;
+            }
+        }
+        &:focus,
+        &.focus {
+            &:after {
+                border-color: v(--link-color-rgb);
+                border-width: 2px;
+            }
+        }
+        &:active,
+        &.active {
+            &:before {
+                opacity: 0.12;
+            }
+        }
+    
+        .Button_icon {
+            color: v(--button-bg-rgb);
+        }
+    }
+
+    &.Button___tertiary:not([disabled]) {
+        &::after {
+            border-color: transparent;
+        }
+        &:focus
+        &.focus {
+            &::after {
+                border-color: v(--button-bg-rgb);
+                border-width: 2px;
+            }
+        }
+    }
+
+    // sizes (small, large, full width)
+    &___small {
+        padding: 10px 12px;
+        height: 32px;
+    }
+    &___large {
+        padding: 14px 20px;
+        height: 48px;
+    }
+    &___fullWidth {
+        width: 100%;
+
+        .Button_icon {
+            flex-shrink: 1;
+        }
+        .Button_label {
+            flex-grow: 1;
+        }
+    }
+
+    // animation
+    .enable-animations & {
+        &::before {
+            transition: opacity $animation-speed-shorter 0s ease-in-out;
```

Comment:
FYI, these sass variables don't yet exist, so we'll need to create them. These are what I started with:

```scss
$animation-speed-shortest: 0.1s;
$animation-speed-shorter: 0.15s;
$animation-speed-short: 0.2s;
$animation-speed-normal: 0.3s;
$animation-speed-long: 0.4s;
$animation-speed-longer: 0.45s;
$animation-speed-longest: 0.5s;
```

The values are somewhat arbitrary and I'm open to thoughts on naming. I basically just figured we should have some variables with purposely defined animation lengths to try maintain consistency across the app as we do more micro-animations and animated transitions.

Reply:
That's a great start! We do have a lot of variables and constants to consider. I was thinking of creating this file once we merge some components together, figure our common CSS and constant variables, like the Icons that are used in both buttons, it would just make it easier to build the picture at least for me, but we can also do it beforehand?

Reply:
After merge is fine!